covquestions-js/src/testCaseRunner/randomTestCaseRunner.test.ts

```diff
@@ -204,4 +204,81 @@ describe("testCaseRunner", () => {
       });
     });
   });
+
+  describe("bug: convert_to_date_string is undefined cases", () => {
+    it("should succeed in random mode if there are no results given", () => {
+      const testQuestionnaire: Questionnaire = {
+        id: "testQuestionnaire",
+        schemaVersion: "1",
+        version: 1,
+        language: "en",
+        title: "All question types",
+        meta: {
+          author: "Someone",
+          availableLanguages: ["en"],
+          creationDate: "2020-04-13T13:48:48+0000",
+        },
+        questions: [
+          {
+            id: "q_contact",
+            text: "Select one Option",
+            type: "select",
+            options: [
+              {
+                text: "yes",
+                value: "yes",
+              },
+              {
+                text: "no",
+                value: "no",
+              },
+            ],
+          },
+          {
+            id: "dateQuestion",
+            text: "Select a date",
+            type: "date",
+            enableWhenExpression: {
+              var: "q_contact.option.yes",
+            },
+          },
+        ],
+        variables: [
+          {
+            id: "v_convert_date",
+            expression: {
+              convert_to_date_string: [
+                {
+                  var: "dateQuestion",
+                },
+                "YYYY.MM.DD",
+              ],
+            },
+          },
+        ],
+        resultCategories: [],
+        testCases: [],
+      };
+
+      const testCase: TestCase = {
+        description: "Test should randomly select values",
+        answers: {
+          dateQuestion: 1617926400,
```

Comment:
I would argue that this is a case that we should not support, since it does not make sense, and it is good that the random test case fails sometimes.

Giving the answer to the date question in the test case but setting the answer to the contact question randomly does not make sense if we need to assert the converted date variable. Also in executions of the questionnaire with real users the converted date will only be set if the contact question was set to yes.

=> I think this is the expected behavior at the moment.

Reply:
Yes, seems like it really is expected. 
I would add a test to preserve the current behaviour, as it definitely makes sense. 

How do we do the test so it does not end up flaky?

Reply:
Are you thinking in the lines of: the errorMessage should have the format "Failed XX out of 100 times. First failure: Wrong variables: expected 'v_convert_date' to be '2021.04.09' but it is 'undefined'", where XX is between 20 and 80?

Theoretically, I could be flaky, but practically, I assume it should be quite stable, if we select 20/80 far away fro 50/50.

Reply:
yes, seems like we should just set the number of runs high enough